cmd/template-resolver/utils/templateResolver.go

```diff
@@ -152,6 +151,33 @@ func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName string) ([
 		return nil, fmt.Errorf("failed to instantiate the template resolver: %w", err)
 	}
 
+	switch kind := policy.GetKind(); kind {
```

Comment:
Question: Does this not work without an assignment?
```suggestion
	switch policy.GetKind() {
```

Reply:
Good catch!